lib/bing_translator.rb

```diff
@@ -143,6 +146,33 @@ def balance
   # Returns existing @access_token if we don't need a new token yet,
   # or returns the one just obtained.
   def get_access_token
+    if @subscription_key
+      get_cognitive_access_token
+    else
+      get_datamarket_access_token
+    end
+  end
+
+private
+
+  def get_cognitive_access_token
+    headers = {
+      'Ocp-Apim-Subscription-Key' => @subscription_key
+    }
+
+    http = Net::HTTP.new(@cognitive_token_uri.host, @cognitive_token_uri.port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE if @skip_ssl_verify
+
+    response = http.post(@cognitive_token_uri.path, "", headers)
+
+    @access_token = {
+      'access_token' => response.body,
+      'expires_at' => Time.now + 480
```

Comment:
Well, I'm not sure how it should be chosen in the first place. I was wondering how you arrived at the current value.